appinventor/blocklyeditor/src/versioning.js

```diff
@@ -743,6 +743,28 @@ Blockly.Versioning.v17_translateComponentSetGetProperty = function(blockElem) {
     mutationElement.removeAttribute('yailtype');
 };
 
+/******************************************************************************
+ Upgrade screen names to use dropdown block.
+ ******************************************************************************/
+
+Blockly.Versioning.makeScreenNamesBeDropdowns = function () {
+  return function(blocksRep, workspace) {
+    var dom = Blockly.Versioning.ensureDom(blocksRep);
```

Comment:
I'm curious about the motivation to use the DOM representation instead of the workspace here. It seems like it would be straightforward to iterate over all of the blocks looking for instances of `controls_openAnotherScreen` and `controls_openAnotherScreenWithStartValue` and replace the value input, if it's of type `text`, with the corresponding option block.

Reply:
Yeah this could be done completely with the workspace representation. The reason I went with dom was just to be consistent with the other upgraders I've written. The method and setter upgraders have to be in dom form because that's what the 'findAllX' functions expect.

I can change it over if you want :D

Reply:
It's fine to leave it as is, I just expected give your comfort with the Blockly API that approach would have been faster to implement.